timed/employment/serializers.py

```diff
@@ -214,88 +228,176 @@ class Meta:
-            else date.today()
-        )
+    def get_credit(self, instance):
+        """Calculate the total credited days.
 
-        reports = Absence.objects.filter(
-            user=instance.user,
-            type=instance.absence_type,
-            date__gte=instance.date,
-            date__lte=end_date
-        ).values_list('duration', flat=True)
+        :return: The total credited days
+        :rtype:  int
+        """
+        if instance.fill_worktime:
+            return None
 
-        return sum(reports, timedelta())
+        start, end = self.get_dates(instance)
 
-    def get_balance_raw(self, instance):
-        """Calculate the balance of the requested credit.
+        return sum(models.AbsenceCredit.objects.filter(
+            user_id=instance.user_id,
+            absence_type=instance,
+            date__gte=start,
+            date__lte=end
+        ).values_list('days', flat=True))
 
-        This is the difference between the credits duration and the total used
-        time.
+    def get_used_days(self, instance):
+        """Calculate the total used days.
 
-        :return: The balance
-        :rtype:  datetime.timedelta
+        :return: The total used days
+        :rtype:  int
         """
-        return (
-            instance.duration - self.get_used_raw(instance)
-            if instance.duration
-            else None
-        )
+        if instance.fill_worktime:
+            return None
 
-    def get_used(self, instance):
-        """Format the total of used time.
+        start, end = self.get_dates(instance)
 
-        :return: The formatted total of used time
+        return len(Absence.objects.filter(
+            user_id=instance.user_id,
+            type=instance,
+            date__gte=start,
+            date__lte=end
+        ))
+
+    def get_used_duration(self, instance):
+        """Calculate the total used duration.
+
+        This is only calculated for types which fill the worktime such as
+        sickness.
+
+        :return: The total used duration
         :rtype:  str
         """
-        used = self.get_used_raw(instance)
+        if not instance.fill_worktime:
+            return None
 
-        return duration_string(used)
+        start, end = self.get_dates(instance)
 
-    def get_balance(self, instance):
-        """Format the balance.
+        return duration_string(sum(Absence.objects.filter(
+            user_id=instance.user_id,
+            type=instance,
+            date__gte=start,
+            date__lte=end
+        ).values_list('duration', flat=True), timedelta()))
 
-        This is None if we don't have a duration.
+    def get_balance(self, instance):
+        """Calculate the balance of credited and used days.
 
-        :return: The formatted balance
-        :rtype:  str or None
+        :return: The balance
+        :rtype:  int
         """
-        balance = self.get_balance_raw(instance)
+        credit = self.get_credit(instance)
```

Comment:
Currently credit needs to be calculated twice. Once for the serializer and once to calculate the balance.
When calculation of sum will be done on database as described above the credit would only be calculated once.